File: Data Structures & Algorithms/jump-game-ii/submission-2.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        if (nums.size() <= 1) {
            return 0;
        }

        int currentEnd = 0, furthestPos = 0, jumps = 0;

        for (int i = 0; i < nums.size() - 1; i++) {
            furthestPos = max(furthestPos, i + nums[i]);

            if (i == currentEnd) {
                jumps++;
                currentEnd = furthestPos;
            }
        }

        return jumps;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        # Creating a set to help check for 
        # consecutive sequences
        sequenceChecker = set()

        # Adding all numbers to our set
        sequenceChecker.update(nums)

        # Setting up to store the longest sequence we find
        longestCount = 0

        # Begin traversing the list
        for num in nums:
            # Checking if it is a new sequence
            if num - 1 not in sequenceChecker:
                # Setting up our main counter
                counter = 1
                
                # Storing the current number we're on
                current = num

                # Checking if it's part of a consecutive sequence
                while current + 1 in sequenceChecker:
                    # Increment our counter
                    counter += 1

                    # Updating our current number
                    current += 1

                # Updating our longest count if necessary
                longestCount = max(longestCount, counter)

        # Returning the longest count we have found
        return longestCount
                